src/m365/entra/commands/enterpriseapp/enterpriseapp-remove.spec.ts

```diff
@@ -0,0 +1,395 @@
+import { settingsNames } from '../../../../settingsNames.js';
+import aadCommands from '../../aadCommands.js';
+
+describe(commands.ENTERPRISEAPP_REMOVE, () => {
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+  let promptIssued: boolean = false;
+
+  const spAppInfo = {
+    "value": [
+      {
+        "id": "59e617e5-e447-4adc-8b88-00af644d7c92",
+        "appId": "65415bb1-9267-4313-bbf5-ae259732ee12",
+        "displayName": "foo",
+        "createdDateTime": "2021-03-07T15:04:11Z",
+        "description": null,
+        "homepage": null,
+        "loginUrl": null,
+        "logoutUrl": null,
+        "notes": null
+      }
+    ]
+  };
+
+  const deleteRequestStub = (): sinon.SinonStub => {
+    return sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/servicePrincipals/59e617e5-e447-4adc-8b88-00af644d7c92') {
+        return;
+      }
+
+      return new Error('Invalid request');
+    });
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.connection.active = true;
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    sinon.stub(cli, 'promptForConfirmation').callsFake(() => {
+      promptIssued = true;
+      return Promise.resolve(false);
+    });
+    promptIssued = false;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.delete,
+      cli.promptForConfirmation,
+      cli.getSettingWithDefaultValue,
+      cli.handleMultipleResultsFound
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ENTERPRISEAPP_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines alias', () => {
+    const alias = command.alias();
+    assert.notStrictEqual(typeof alias, 'undefined');
+  });
+
+  it('defines correct alias', () => {
+    const alias = command.alias();
+    assert.deepStrictEqual(alias, [aadCommands.SP_REMOVE, commands.SP_REMOVE]);
+  });
+
+  it('fails when the specified Entra app does not exist', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/servicePrincipals?$filter=displayName eq 'Invalid'`) {
+        return {
+          "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#servicePrincipals",
```

Comment:
Since we're removing the odata metadata, this property shouldn't be in the request.